capstone/src/main/java/com/google/univiz/api/SearchResource.java

```diff
@@ -1,7 +1,22 @@
 package com.google.univiz.api;
 
+import java.util.List;
+
 /**
  * The SearchResoure interface is responsible for all data management/processing specific for the
  * search feature of the UniViz webapp.
  */
-public interface SearchResource {}
+public interface SearchResource {
+  /**
+   * Given a potentially incomplete search query, returns a list of colleges that would match the
+   * query.
+   *
+   * @param query The search query as the user typed in. It should be a non-empty and a non-null
+   *     string.
```

Comment:
A few nits on this param doc:
* "as the user typed in" - in the scope of this project, this will probably be the case, but we could imagine other uses for this interface (ie a batch job trying to match text).
* what would the consequences be of invoking this method with an empty or null string? we have a couple of options here:
  - document what happens in this situations
  - be defensive so the client doesn't need to worry about this (ie empty input yields empty results).

Reply:
I would probably go with the defensive strategy